libraries/AP_BattMonitor/AP_BattMonitor_MAVLink_Battery.cpp

```diff
@@ -0,0 +1,129 @@
+#include <AP_HAL/AP_HAL.h>
+#include <GCS_MAVLink/GCS_MAVLink.h>
+#include "AP_BattMonitor_MAVLink_Battery.h"
+
+AP_BattMonitor_MAVLink_Battery::AP_BattMonitor_MAVLink_Battery(AP_BattMonitor &mon,
+                                             AP_BattMonitor::BattMonitor_State &mon_state,
+                                             AP_BattMonitor_Params &params) :
+    AP_BattMonitor_Backend(mon, mon_state, params)
+{
+	_state.healthy = false;
+}
+
+void AP_BattMonitor_MAVLink_Battery::read()
+{
+    if ((AP_HAL::micros() - _state.last_time_micros) > MAVLINK_SMART_BATTERY_TIMEOUT_MICROS) {
+		_state.healthy = false;
+		return;
+	} 
+	_state.healthy = true;	
+   return;
+}
+
+bool AP_BattMonitor_MAVLink_Battery::capacity_remaining_pct(uint8_t &cap_remain_pct) const
+{
+    if (capacity_rem_pct != INT8_MIN) {
+		/* positive capacity remaining is expected */
+		if (capacity_rem_pct > 0) {
+			cap_remain_pct = capacity_rem_pct;
+		} else {
+			cap_remain_pct = 0;
+		}
+    	return true;
+	}
+	return false;
+}
+
+bool AP_BattMonitor_MAVLink_Battery::get_cycle_count(uint16_t &cycles) const
+{
+    if (cycle_count != UINT16_MAX) {
+		cycles = cycle_count;
+    	return true;
+	}
+	return false;
+}
+
+void AP_BattMonitor_MAVLink_Battery::process_info_msg(const mavlink_message_t &msg)
+{
+	int32_t capacity = mavlink_msg_smart_battery_info_get_capacity_full(&msg);

```

Comment:
This is for a follow-up PR, but it might be worthwhile to add a battery options bit to allow this packet variable.discharge_minimum_voltage to control the battery critical voltage params? Automatically configures itself that way!

Reply:
i just added it, because it could be useful if my coloumb counter fails; I use 2 hall sensors supplied by a measured 5V rail with  a 3V3 ADC reference and I calculate the current consumption quite accurately - also over longer time periods; therefore a capacity driven failsafe is highly recommended for my product; i don't want to give somebody the impression this would be a correct failsafe value for high and low current consumption;  i just noticed that there is a 'set_default' option, which might be fine, if it just sets the value if unconfigured;